waku/v2/node/jsonrpc/filter_api.nim

```diff
@@ -16,10 +17,20 @@ proc installFilterApiHandlers*(node: WakuNode, rpcsrv: RpcServer) =
     messageCache = initTable[ContentTopic, seq[WakuMessage]]()
   
   proc filterHandler(msg: WakuMessage) {.gcsafe, closure.} =
-    debug "WakuMessage received", msg=msg
     # Add message to current cache
-    # @TODO limit max content topics and messages
-    messageCache.mgetOrPut(msg.contentTopic, @[]).add(msg)
+    trace "WakuMessage received", msg=msg
+    
+    # Make a copy of msgs for this topic to modify
+    var msgs = messageCache.getOrDefault(msg.contentTopic, @[])
```

Comment:
Why not changing `messageCache` directly without copying?

Reply:
This makes it a bit easier to avoid concurrent modification errors if `messageCache` can be written to/read from multiple threads (say a `handler` inserting new messages while the cache is being read). No real benefit as it stands, but in future I would like to move `messageCache` to a more sensible place where it could theoretically be accessible from other threads.